public/public.class.php

```diff
@@ -215,15 +211,116 @@ public static function listing(){
    *
    * @since    1.0.0
    */
-  public static function listing_grid(){
+  public static function listing_grid($opts){
+
+    $layout = empty($opts['layout']) ? 1 : $opts['layout'];
+
+    self::$options = get_option( 'purecharity_fundraisers_settings' );
+
+    switch ((int) $layout) {
+      case 1:
+        return self::grid_option_3();
+        break;
```

Comment:
Is the grid_option_3 tied to option 1 or is it grid_option_1 here?  Trying to make sure I am walking through the changes since I don't have a local pull of it right now.